pauthabielf64/pauthabielf64.rst

```diff
@@ -1103,80 +1106,94 @@ AUTH variant GOT Generating Relocations
+  +-------------+----------------------------------------+----------------------------------------+--------------------------+
+  | 0x811A      | R\_AARCH64\_AUTH\_LD64\_GOT\_LO12_NC   | G(ENCD(GDAT(S)))                       | Set the LD/ST immediate  |
+  |             |                                        |                                        | field to bits [11:3] of  |
+  |             |                                        |                                        | X. No overflow check;    |
+  |             |                                        |                                        | check that X&7 = 0       |
+  +-------------+----------------------------------------+----------------------------------------+--------------------------+
+  | 0x811B      | R\_AARCH64\_AUTH\_LD64\_GOTPAGE\_LO15  | G(ENCD(GDAT(S))) - Page(GOT)           | Set the LD/ST immediate  |
+  |             |                                        |                                        | field to bits [14:3] of  |
+  |             |                                        |                                        | X; check that 0 <= X  <  |
+  |             |                                        |                                        | 2\ :sup:`15`             |
+  +-------------+----------------------------------------+----------------------------------------+--------------------------+
+  | 0x811C      | R\_AARCH64\_AUTH\_GOT\_ADD_LO12_NC     | G(ENCD(GDAT(S)))                       | Set an ADD immediate     |
+  |             |                                        |                                        | value to bits [11:0] of  |
+  |             |                                        |                                        | X. No overflow check.    |
+  +-------------+----------------------------------------+----------------------------------------+--------------------------+
+  | 0x811D      | R\_AARCH64\_AUTH\_GOT\_ADR\_PREL\_LO21 | G(ENCD(GDAT(S))) - P                   | Set the immediate        |
+  |             |                                        |                                        | value to bits[20:0] of X;|
+  |             |                                        |                                        | check that -2 :sup:`20`  |
+  |             |                                        |                                        | <= 2 :sup: `20`          |
+  +-------------+----------------------------------------+----------------------------------------+--------------------------+
+  | 0x811E      | R\_AARCH64\_AUTH\_TLSDESC\_ADR\_PAGE21 | Page(G(ENCD(GTLSDESC(S)))) - Page(P)   | Set an ADRP immediate    |
+  |             |                                        |                                        | field to bits [32:12] of |
+  |             |                                        |                                        | X; check –2\ :sup:`20`   |
```

Comment:
The range for ADRP is ±4GB [1], so I think that the check should be between -2^32 and 2^32, instead of 20.

[1] https://developer.arm.com/documentation/100076/0100/A64-Instruction-Set-Reference/A64-General-Instructions/ADRP

Reply:
ACK. It looks like I used the text from `R_<CLS>_TLSDESC_ ADR_PREL21` in the table rather than the similarly named R_<CLS>_TLSDESC_ ADR_PAGE21

Will fix.

Reply:
I've updated the PR to use -2^32 and 2^32